crates/extensions/c8y_mapper_ext/src/actor.rs

```diff
@@ -60,15 +64,36 @@ pub(crate) type IdDownloadRequest = (CmdId, DownloadRequest);
 fan_in_message_type!(C8yMapperInput[MqttMessage, FsWatchEvent, SyncComplete, IdUploadResult, IdDownloadResult] : Debug);
 type C8yMapperOutput = MqttMessage;
 
+#[derive(Clone)]
+struct MqttPublisher(Arc<Mutex<LoggingSender<MqttMessage>>>);
```

Comment:
Any specific reason you couldn't just freely clone the senders and use them independently? I see that you're already planning on exploring that option, but just wondering why you chose this path first. Faced any issues with cloned senders?

Reply:
I did that first because I knew that I'll need `Mutex`es for synchronization of other parts, and to be honest, I forgot that our senders are `Clone`, so I thought `Mutex` is the only option, but it's not the case.